### PR TITLE
[AN-5062] First version: Previews for Soundcloud and Spotify links. 

### PR DESCRIPTION
**Pull request by makingthematrix**
New *PartViews and layouts for both. And since they're very similar - a common abstract class.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Can one of the admins verify this patch?

**Comment by JPonte** [comment]
build again

File: app/src/main/scala/com/waz/zclient/messages/MessageViewFactory.scala [hunk 1]
```diff
@@ -62,7 +61,8 @@ class MessageViewFactory {
         case ConnectRequest => ViewHelper.inflate(R.layout.message_connect_request, parent, false)
         case InviteBanner   => ViewHelper.inflate(R.layout.message_invite_banner, parent, false)
         case OtrMessage     => ViewHelper.inflate(R.layout.message_otr_part, parent, false)
-        case SoundCloud     => ViewHelper.inflate(R.layout.message_text, parent, false) // TODO: implement media player
+        case SoundCloud     => ViewHelper.inflate(R.layout.message_soundcloud, parent, false) // TODO: implement media player
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/messages/MessageViewFactory.scala]: the TODO aren't needed anymore

File: app/src/main/scala/com/waz/zclient/messages/parts/SoundPreviewPartView.scala [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  */
+abstract class SoundPreviewPartView(context: Context, attrs: AttributeSet, style: Int, messagePartType: Message.Part.Type)
+  extends RelativeLayout(context, attrs, style) with ClickableViewPart with ViewHelper {
+
+  def getImageView: ImageView
+  def getTitleView: TextView
+  def getArtistView: TextView
+  def getErrorView: GlyphTextView
+
+  //val network = inject[NetworkModeService] // currently not used, but network.networkMode might be useful
```
Comment (JPonte) [inline app/src/main/scala/com/waz/zclient/messages/parts/SoundPreviewPartView.scala]: No need to keep this comment.

**Pull request closed without merging**